src/url.js

```diff
@@ -59,7 +59,28 @@ export function parseUrl(url, opt_nocache) {
   if (fromCache) {
     return fromCache;
   }
+
+  const info = parseUrlWithA(a, url);
+
+  // Freeze during testing to avoid accidental mutation.
+  const frozen = (getMode().test && Object.freeze) ? Object.freeze(info) : info;
+
+  if (opt_nocache) {
+    return frozen;
+  }
+  return cache[url] = frozen;
+}
+
+/**
+ * Returns a Location-like object for the given URL. If it is relative,
+ * the URL gets resolved.
+ * @param {!HTMLAnchorElement} a
+ * @param {string} url
+ * @return {!Location}
+ */
+export function parseUrlWithA(a, url) {
```

Comment:
Not dangerous, but we have a cached anchor in this module to avoid every ole caller making there own. And I'd probably want to review anyone using this to make sure they intend to use a cross-origin anchor element.